### PR TITLE
Create Volunteer Profile Side Panel

### LINKED ISSUE #505: Admin schedule page profile button should bring up user profile sidebar
When clicking on profile info button on admin schedule page, it should bring up the user profile sidebar for that user! 

Expected:

Figma: https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=888%3A18932

Technical
- You will add a new sidebar component which will open when you click the profile icon on the posting schedule sidebar
- The relevant route to be changed is `admin/schedule/posting/:id`

### PR DESCRIPTION
**Pull request by twilkhoo**
## Ticket link

Closes #505 

## Implementation description
* Created the volunteer profile side bar
* Used `volunteerUserById` and `getShiftSignupsForUser` queries to get all the info needed

image

With local repo seed:
image

## Steps to test
1. Navigate to `.../admin/schedule/posting/1` (or /2), click on a shift that has available volunteers, and then click the profile icon of a volunteer
2. Ensure the site doesn't crash, and navigate between volunteers and shifts (the side panel should either display volunteer profiles or shift information)

## What should reviewers focus on?
* Review implementation logic
* Review empty state design implementations (i.e. when there are no skills listed, or no shifts of a certain type)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/prisma/seed.ts [hunk 1]
```diff
@@ -261,6 +268,7 @@ const main = async () => {
             create: {
               ...user.volunteer,
               branches: { connect: user.volunteer.branches },
+              skills: { connect: user.volunteer.skills },
```
Comment (MatoPlus) [inline backend/prisma/seed.ts:271]: Thanks for adding the seed!

**Review by MatoPlus — COMMENTED** [review]
There seems to be quite a few linting issues but it looks good on the surface. Will take a closer look later today!

File: frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        borderBottom="1px"
+        borderColor="background.dark"
+      >
+        <Text textStyle="body-bold">Requested Shifts</Text>
+      </VStack>
+
+      <VStack w="full" spacing={0} overflow="auto">
+        {shifts?.filter((shift) => shift.status === "PENDING").length > 0 ? (
+          shifts
+            ?.filter((shift) => shift.status === "PENDING")
```
Comment (MatoPlus) [inline frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx:41]: optional: is it possible to remove this redundancy? Maybe make this an inline var of this code block or use state + useEffect

**Review by MatoPlus — APPROVED** [review]
This looks good to me with optional nit. Thanks for getting this done!

File: frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        borderBottom="1px"
+        borderColor="background.dark"
+      >
+        <Text textStyle="body-bold">Scheduled Shifts</Text>
+      </VStack>
+
+      <VStack w="full" spacing={0} overflow="auto">
+        {shifts?.filter((shift) => shift.status === "CONFIRMED").length > 0 ? (
+          shifts
+            ?.filter((shift) => shift.status === "CONFIRMED")
```
Comment (MatoPlus) [inline frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx:74]: idem

File: frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx (hunk 2 above)
Comment (twilkhoo) [reply frontend/src/components/admin/schedule/volunteersidepanel/VolunteerShiftsTable.tsx]: Good point, useState and useEffect did the trick!

**Comment by twilkhoo** [comment]
In addition to the shift filtering redundancy, I added some logic for when the volunteer panel unmounts, since there were some unmounting warnings without it.

**Review by MatoPlus — APPROVED** [review]
LGTM, thanks for getting this done!

**Pull request merged**